### PR TITLE
CI/CD Integration

### PR DESCRIPTION
**Pull request by sergeuz**
https://app.shortcut.com/particle/story/91839/setup-device-os-test-runner-to-run-linter-tests-in-circleci

This PR adds a CI/CD config for GitHub Actions, enables the linter and adds some rudimentary unit tests.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          - 12
+        optional:
+          - false
+        include:
+          # TODO: particle-usb doesn't support Node.js 14 and higher:
+          # https://github.com/particle-iot/particle-usb/pull/47
+          - node: 14
+            optional: true
+          - node: 16
+            optional: true
```
Comment (joegoggins) [inline .github/workflows/ci.yaml]: Let's comment these lines out so the build starts green on Node 12

**Review by joegoggins — CHANGES_REQUESTED** [review]
Thanks @sergeuz . I took this for a spin and discovered a couple of things. I fixed some (see my commits pushed to branch) and left a couple for you. See [notes](https://app.shortcut.com/particle/story/91839/setup-device-os-test-runner-to-run-linter-tests-in-circleci#activity-92870) for details, and sub-tasks for work remaining. Let me know if you have any questions.

**Comment by laupow** [comment]
@monkbroc drafted a good pipeline for npm packages in Circle mongoose-models PR 273

Is there time this sprint to switch to Circle? Since CI is under maintenance this sprint anyways, this is a good opportunity  to build some mindshare around publishing npm modules in Circle

The only bit to add on top of Julien's PR may be matrix builds: https://circleci.com/blog/circleci-matrix-jobs/. Happy to help get the matrix bit working

**Comment by sergeuz** [comment]
Thanks for letting me know, @laupow. I'll look into moving this repo to CircleCI but I'm not sure when I have time for that. The goal of this PR was not just to add a CI/CD config to this repo but also bring it up to our basic standards in regards to automatic testing, code style and etc. I picked GitHub Actions because I'm familiar with them, but now that the dirty work is done we can switch the CI/CD system used by this repo at any time.

**Review by joegoggins — APPROVED** [review]
Tests are now passing for me. Looks good. LTGM.

@laupow regarding the ask to move this to CircleCI, I enqueued this for later: https://app.shortcut.com/particle/story/93359/migrate-device-os-flash-util-and-device-os-test-runner-to-circleci

Sergey, I don't believe there is fierce urgency on that, so feel free to tackle whenever you have a chance. Given the great CircleCI npm lib reference model that Julien shared in https://github.com/particle-iot/mongoose-models/pull/273 , I don't think it will take more than an hour for both repos.

**Pull request merged**